OpenRA.Game/Actor.cs

```diff
@@ -40,34 +40,38 @@ public sealed class Actor : IScriptBindable, IScriptNotifyBind, ILuaTableBinding
 		public Group Group;
 		public int Generation;
 
-		Lazy<Rectangle> bounds;
-		Lazy<Rectangle> visualBounds;
-		Lazy<IFacing> facing;
-		Lazy<Health> health;
-		Lazy<IOccupySpace> occupySpace;
-		Lazy<IEffectiveOwner> effectiveOwner;
-
-		public Rectangle Bounds { get { return bounds.Value; } }
-		public Rectangle VisualBounds { get { return visualBounds.Value; } }
-		public IOccupySpace OccupiesSpace { get { return occupySpace.Value; } }
-		public IEffectiveOwner EffectiveOwner { get { return effectiveOwner.Value; } }
+		public Rectangle Bounds { get; private set; }
+		public Rectangle VisualBounds { get; private set; }
+		public IEffectiveOwner EffectiveOwner { get; private set; }
+		public IOccupySpace OccupiesSpace
+		{
+			get
+			{
+				if (occupySpace == null)
+					occupySpace = Trait<IOccupySpace>();
+				return occupySpace;
+			}
+		}
```

Comment:
Minor behaviour change here. Accessing this now throws if IOccupySpace is missing, rather than returning null. None of our existing code is affected by this change.